driver-core/src/test/java/com/datastax/driver/core/PagingStateTest.java

```diff
@@ -0,0 +1,109 @@
+package com.datastax.driver.core;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+import com.datastax.driver.core.exceptions.InvalidPagingStateException;
+
+
+public class PagingStateTest extends CCMBridge.PerClassSingleNodeCluster {
+
+    @Override
+    protected Collection<String> getTableDefinitions() {
+        return Collections.singletonList("CREATE TABLE test (k text, v int, PRIMARY KEY (k, v))");
+    }
+
+    @Test(groups = "short")
+    public void changeInPagingStateFailsVerification (){
+
+        String key = "paging_test";
+        for (int i = 0; i < 100; i++)
+            session.execute(String.format("INSERT INTO test (k, v) VALUES ('%s', %d)", key, i));
```

Comment:
You should move this to a `BeforeClass`-annotated method, since all your tests depend on it.